packages/widget-v2/src/pages/SwapPage/useUpdateAmountWhenRouteChanges.ts

```diff
@@ -17,18 +18,24 @@ export const useUpdateAmountWhenRouteChanges = () => {
 
   useEffect(() => {
     if (!route.data || !sourceAsset || !destinationAsset) return;
+    if (sourceAsset?.amount === "" && direction === "swap-in") return;
+    if (destinationAsset?.amount === "" && direction === "swap-out") return;
```

Comment:
this is needed to allow us to properly clear inputs, otherwise it gets immediately overridden by the code below and sets it to the previously fetched swapInAmount/swapOutAmount